urllib3/response.py

```diff
@@ -56,14 +56,30 @@ class GzipDecoder(object):
 
     def __init__(self):
         self._obj = zlib.decompressobj(16 + zlib.MAX_WBITS)
+        self._state = 'first'
 
     def __getattr__(self, name):
         return getattr(self._obj, name)
 
     def decompress(self, data):
+        ret = binary_type()
+        if self._state == 'error':
+            return ret
         if not data:
             return data
-        return self._obj.decompress(data)
+        try:
+            ret = self._obj.decompress(data)
```

Comment:
Can we formulate the loop in such a way that we don't have to "prime" it with this extra line?

Reply:
Done.